Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -0,0 +1,482 @@
+
+        }
+        public LayoutType Layout
+        {
+            get { return layout; }
+            set { layout = value; }
+
+        }
+        public float Radius
+        {
+            get { return radius; }
+            set { radius = value; }
+
+        }
+        public int Rows
+        {
+            get { return rows; }
+            set { rows = value; }
+
+        }
+        public float CellWidth
+        {
+            get { return cellWidth; }
+            set { cellWidth = value; }
+
+        }
+        public float CellHeight
+        {
+            get { return cellHeight; }
+            set { cellHeight = value; }
+
+        }
+
+        /// <summary>
+        /// Reference mesh to use for rendering the sphere layout
+        /// </summary>
+        public Mesh SphereMesh
+        {
+            get { return sphereMesh; }
+            set { sphereMesh = value; }
+
+        }
+
+        /// <summary>
+        /// Reference mesh to use for rendering the cylinder layout
+        /// </summary>
+        public Mesh CylinderMesh
+        {
+            get { return cylinderMesh; }
+            set { cylinderMesh = value; }
+        }
+
+        public float Width
+        {
+            get { return width; }
+        }
+
+        public float Height
+        {
+            get { return height; }
+        }
+        #endregion
+
+
+        /// <summary>
+        /// Overriding base function function for laying out all the children when UpdateCollection is called.
+        /// </summary>
+        protected override void LayoutChildren()
+        {
+
+            columns = Mathf.CeilToInt((float)NodeList.Count / Rows);
+            width = columns * CellWidth;
+            height = Rows * CellHeight;
+            halfcell = new Vector2(CellWidth / 2f, CellHeight / 2f);
+            circumferenc = 2f * Mathf.PI * Radius;
+
+
+            int cellCounter = 0;
+            float startOffsetX;
+            float startOffsetY;
+
+            Vector3[] nodeGrid = new Vector3[NodeList.Count];
+            Vector3 newPos = Vector3.zero;
+            Vector3 newRot = Vector3.zero;
+
+            // Now lets lay out the grid
+            startOffsetX = (columns * 0.5f) * CellWidth;
+            startOffsetY = (Rows * 0.5f) * CellHeight;
+
+            cellCounter = 0;
+
+            // First start with a grid then project onto surface
+            switch (layout)
+            {
+                case LayoutType.ColumnThenRow:
+                default:
+                    for (int c = 0; c < columns; c++)
+                    {
+                        for (int r = 0; r < Rows; r++)
+                        {
+                            if (cellCounter < NodeList.Count)
+                            {
+                                nodeGrid[cellCounter] = new Vector3((c * CellWidth) - startOffsetX + halfcell.x, -(r * CellHeight) + startOffsetY - halfcell.y, 0f) + (Vector3)((NodeList[cellCounter])).Offset;
```

Comment:
We probably don't need to assign a new vector here, seeing as we create a new grid array anyway.

Reply:
Just assign the x,y,z values